mifospay/src/main/java/org/mifos/mobilewallet/mifospay/bank/ui/BankAccountDetailActivity.kt

```diff
@@ -2,118 +2,52 @@ package org.mifos.mobilewallet.mifospay.bank.ui
 import android.os.Bundle
-import android.view.View
-import android.widget.Button
-import android.widget.TextView
-import androidx.appcompat.widget.Toolbar
-import androidx.cardview.widget.CardView
-import butterknife.BindView
-import butterknife.ButterKnife
-import butterknife.OnClick
-import dagger.hilt.android.AndroidEntryPoint
+import androidx.activity.compose.setContent
 import com.mifos.mobilewallet.model.domain.BankAccountDetails
-import org.mifos.mobilewallet.mifospay.R
-import org.mifos.mobilewallet.mifospay.bank.BankContract
-import org.mifos.mobilewallet.mifospay.bank.BankContract.BankAccountDetailView
-import org.mifos.mobilewallet.mifospay.bank.presenter.BankAccountDetailPresenter
+import dagger.hilt.android.AndroidEntryPoint
 import org.mifos.mobilewallet.mifospay.base.BaseActivity
 import org.mifos.mobilewallet.mifospay.common.Constants
-import org.mifos.mobilewallet.mifospay.utils.DebugUtil
+import org.mifos.mobilewallet.mifospay.theme.MifosTheme
 import org.mifos.mobilewallet.mifospay.utils.Toaster
-import javax.inject.Inject
 
 @AndroidEntryPoint
-class BankAccountDetailActivity : BaseActivity(), BankAccountDetailView {
-    @JvmField
-    @Inject
-    var mPresenter: BankAccountDetailPresenter? = null
-    var mBankAccountDetailPresenter: BankContract.BankAccountDetailPresenter? = null
-
-    @JvmField
-    @BindView(R.id.toolbar)
-    var mToolbar: Toolbar? = null
-
-    @JvmField
-    @BindView(R.id.tv_bank_name)
-    var mTvBankName: TextView? = null
-
-    @JvmField
-    @BindView(R.id.tv_account_holder_name)
-    var mTvAccountHolderName: TextView? = null
-
-    @JvmField
-    @BindView(R.id.tv_branch)
-    var mTvBranch: TextView? = null
-
-    @JvmField
-    @BindView(R.id.tv_ifsc)
-    var mTvIfsc: TextView? = null
-
-    @JvmField
-    @BindView(R.id.tv_type)
-    var mTvType: TextView? = null
-
-    @JvmField
-    @BindView(R.id.btn_setup_upi_pin)
-    var mBtnSetupUpiPin: Button? = null
-
-    @JvmField
-    @BindView(R.id.cv_change_upi_pin)
-    var mCvChangeUpiPin: CardView? = null
-
-    @JvmField
-    @BindView(R.id.cv_forgot_upi_pin)
-    var mCvForgotUpiPin: CardView? = null
+class BankAccountDetailActivity : BaseActivity() {
 
-    @JvmField
-    @BindView(R.id.btn_delete_bank)
-    var mBtnDeleteBank: Button? = null
-    private var bankAccountDetails: BankAccountDetails? = null
+    private lateinit var bankAccountDetails: BankAccountDetails
     private var index = 0
+
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
-        setContentView(R.layout.activity_bank_account_detail)
-        ButterKnife.bind(this)
-        showColoredBackButton(R.drawable.ic_arrow_back_black_24dp)
-        setToolbarTitle(Constants.BANK_ACCOUNT_DETAILS)
-        mPresenter!!.attachView(this)
-        bankAccountDetails = intent.extras!!.getParcelable(Constants.BANK_ACCOUNT_DETAILS)
+        bankAccountDetails = intent.extras?.getParcelable(Constants.BANK_ACCOUNT_DETAILS)!!
```

Comment:
Don't pass the bank details from intent instead pass the bank ID and create ViewModel for bank details and load bank detail from Id.